### PR TITLE
Updating torchvision tutorial using Transforms API v2

### PR DESCRIPTION
**Pull request by vfdev-5**
## Description
- Updating torchvision tutorial using Transforms API v2

TODO:
- [x] fix exec error with COCO eval (https://github.com/pytorch/vision/pull/7869)
- [x] update _static/tv-training-code.py
- [x] update _static/torchvision_finetuning_instance_segmentation.ipynb or remove as unused and outdated? => decided to skip this and opt for sphinx gallery way of doing things.

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

cc @datumbox @nairbv @fmassa @NicolasHug @YosuaMichael

### CODE REVIEW

File: intermediate_source/torchvision_tutorial.rst [hunk 1]
```diff
@@ -103,13 +106,21 @@ Here is one example of a pair of images and segmentation masks
 So each image has a corresponding
 segmentation mask, where each color correspond to a different instance.
 Let’s write a ``torch.utils.data.Dataset`` class for this dataset.
+In the code below we are wrapping images, bounding boxes and masks into
+``torchvision.datapoints`` structures such that we will be able apply torchvision
+built-in transformations (`new Transforms API <https://pytorch.org/vision/stable/transforms.html>`_)
+that cover the object detection and segmetation tasks.
+More on torchvision datapoints see this `documentation <https://pytorch.org/vision/stable/datapoints.html>`_
```
Comment (svekars) [inline intermediate_source/torchvision_tutorial.rst:113]:
```suggestion
In the code below, we are wrapping images, bounding boxes and masks into
``torchvision.datapoints`` structures so that we will be able to apply torchvision
built-in transformations (`new Transforms API <https://pytorch.org/vision/stable/transforms.html>`_)
that cover the object detection and segmetation tasks.
For more information about torchvision datapoints, see `this documentation <https://pytorch.org/vision/stable/datapoints.html>`_
```

File: intermediate_source/torchvision_tutorial.rst [hunk 2]
```diff
@@ -103,13 +98,21 @@ Here is one example of a pair of images and segmentation masks
 So each image has a corresponding
 segmentation mask, where each color correspond to a different instance.
 Let’s write a ``torch.utils.data.Dataset`` class for this dataset.
+In the code below, we are wrapping images, bounding boxes and masks into
+``torchvision.datapoints`` structures so that we will be able to apply torchvision
+built-in transformations (`new Transforms API <https://pytorch.org/vision/stable/transforms.html>`_)
+that cover the object detection and segmentation tasks.
+For more information about torchvision datapoints see `this documentation <https://pytorch.org/vision/stable/datapoints.html>`_.
```
Comment (NicolasHug) [inline intermediate_source/torchvision_tutorial.rst:105]: I think we should link to https://pytorch.org/vision/main/auto_examples/v2_transforms/plot_transforms_v2.html#sphx-glr-auto-examples-v2-transforms-plot-transforms-v2-py instead, as there isn't much relevant info in https://pytorch.org/vision/stable/datapoints.html.

File: intermediate_source/torchvision_tutorial.rst [hunk 3]
```diff
@@ -2,17 +2,17 @@ TorchVision Object Detection Finetuning Tutorial
...
 
 For this tutorial, we will be finetuning a pre-trained `Mask
-R-CNN <https://arxiv.org/abs/1703.06870>`__ model in the `Penn-Fudan
+R-CNN <https://arxiv.org/abs/1703.06870>`__ model on the `Penn-Fudan
 Database for Pedestrian Detection and
 Segmentation <https://www.cis.upenn.edu/~jshi/ped_html/>`__. It contains
 170 images with 345 instances of pedestrians, and we will use it to
 illustrate how to use the new features in torchvision in order to train
-an instance segmentation model on a custom dataset.
+an object detection model on a custom dataset.
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply intermediate_source/torchvision_tutorial.rst:15]: Well, yes mask-rcnn is about object detection and instance segmentation together and it is generalization of an object detection architecture. How about saying there:
```
how to use the new features in torchvision in order to train
an object detection and instance segmentation model on a custom dataset.
```
?

[the remark this answers is not in the document]
Comment (NicolasHug) [reply intermediate_source/torchvision_tutorial.rst:15]: sure

File: intermediate_source/torchvision_tutorial.rst [hunk 4]
```diff
@@ -24,44 +24,39 @@ adding new custom datasets. The dataset should inherit from the standard
 ``__getitem__``.
 
 The only specificity that we require is that the dataset ``__getitem__``
-should return:
+should return a tuple:
 
--  image: a PIL Image of size ``(H, W)``
+-  image: ``torchvision.datapoints.Image[3, H, W]`` or a PIL Image of size ``(H, W)``
```
[the remark this answers is not in the document]
Comment (vfdev-5) [reply intermediate_source/torchvision_tutorial.rst:29]: It works locally :+1: 
Let's try it here

File: intermediate_source/torchvision_tutorial.rst [hunk 5]
```diff
@@ -24,44 +24,40 @@ adding new custom datasets. The dataset should inherit from the standard
 ``__getitem__``.
 
 The only specificity that we require is that the dataset ``__getitem__``
-should return:
+should return a tuple:
 
--  image: a PIL Image of size ``(H, W)``
+-  image: :class:`torchvision.datapoints.Image` of shape ``[3, H, W]`` or a PIL Image of size ``(H, W)``
```
Comment (NicolasHug) [inline intermediate_source/torchvision_tutorial.rst:29]: Do we want to indicate that a Tensor is fine as well?

```suggestion
-  image: :class:`torchvision.datapoints.Image` of shape ``[3, H, W]``, a pure tensor, or a PIL Image of size ``(H, W)``
```

File: intermediate_source/torchvision_tutorial.rst [hunk 6]
```diff
@@ -94,22 +90,34 @@ have the following folder structure:
...
 segmentation mask, where each color correspond to a different instance.
-Let’s write a ``torch.utils.data.Dataset`` class for this dataset.
+Let’s write a :class:`torch.utils.data.Dataset` class for this dataset.
+In the code below, we are wrapping images, bounding boxes and masks into
+``torchvision.datapoints`` classes so that we will be able to apply torchvision
+built-in transformations (`new Transforms API <https://pytorch.org/vision/stable/transforms.html>`_)
+for the given object detection and segmentation task.
+Namely, image tensors will be wrapped by :class:`torchvision.datapoints.Image`, bounding boxes into
+:class:`torchvision.datapoints.BoundingBoxes` and masks into :class:`torchvision.datapoints.Mask`.
+As datapoints are :class:`torch.Tensor` subclasses, wrapped objects are also tensors and inherit plain
```
Comment (NicolasHug) [inline intermediate_source/torchvision_tutorial.rst:108]:
```suggestion
As datapoints are :class:`torch.Tensor` subclasses, wrapped objects are also tensors and inherit the plain
```

**Review by NicolasHug — APPROVED** [review]
Thanks @vfdev-5 , the changes LGTM

**Comment by vfdev-5** [comment]
@svekars we are planning to rewrite this tutorial as a python file (same as a lot of other tutorials, e.g. https://github.com/pytorch/tutorials/blob/main/intermediate_source/spatial_transformer_tutorial.py) and rely on sphinx gallery to render the content and ipynb file. 
Do you have any thoughts / suggestions about that, also concerning CI part ? For this tutorial we need to download the dataset, model weights and run the training/evaluation. Locally the training on recent GPUs takes less than 5 minutes (we can also reduce the number of epochs).
